parlai/tasks/amazon_qa/agents.py

```diff
@@ -26,11 +26,14 @@ def __init__(self, opt, shared=None):
             self._setup_data()
         self.reset()
 
+        self.num_ex = sum([len(x) for x in self.data])
```

Comment:
this can also be slow because you're calculating it for every teacher in the batch. might want to just do it in the main teacher and "share" the number.